src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -133,6 +133,33 @@ preflight1(PreflightContext const& ctx)
 NotTEC
 preflight2(PreflightContext const& ctx)
 {
+    if (ctx.flags & tapDRY_RUN)  // simulation
+    {
+        if (ctx.tx.getSignature().empty())
```

Comment:
Could you please rewrite the logic with early returns, it is hard to follow with many section included
Like:

```
if (!ctx.tx.getSignature().empty())
    return temINVALID;
if (!ctx.tx.isFieldPresent(sfSigners))
    return tesSUCCESS;
    ...
```

Reply:
fixed in 316639642e6a4d8111a5da29af35c965dbc11d01